.github/workflows/ci-core.yml

```diff
@@ -88,15 +88,21 @@ jobs:
         type:
           - cmd: go_core_tests
             id: core_unit
+            os: ubuntu-latest-32cores-128GB
+          - cmd: go_core_tests_short
+            id: core_unit_short
+            os: ubuntu-latest-4cores-16GB
           - cmd: go_core_race_tests
             id: core_race
+            os: ubuntu-latest-16cores-64GB
```

Comment:
Having a higher number of cores will improve race detection.

Reply:
Ok - ideally we avoid the 64-core variant. Do you think 32-cores is enough? That would still bring many cost savings.

Reply:
It's always a sliding scale, but FWIW many races get past this check on PRs and are only caught during overnight runs.

Reply:
Ok - in that case, it sounds worthwhile to implement a condition which uses the 64-core variant for the overnight runs. If that's the case, do you still recommend 32-cores for day-to-day PR runs?

Reply:
What was it using before?

Reply:
There are other parameters that could be adjusted too. The race step is meant to "soak" for as long as the normal tests are running, by adjusting the -count and -timeout.

Reply:
All `core_race` tests are currently using 64cores.

Reply:
Discussed offline.
If we need to up this to 64 cores again (for PRs) due to issues, then we can cross that bridge when we get there.